ocaml-lsp-server/src/ocaml_lsp_server.ml

```diff
@@ -341,6 +346,46 @@ let hover (state : State.t) { HoverParams.textDocument = { uri }; position } =
     let resp = Hover.create ~contents ~range () in
     Ok (Some resp, state)
 
+let signature_help (state : State.t)
+    { SignatureHelpParams.textDocument = { uri }; position; context = _ } =
+  let store = state.store in
+  let uri = Uri.t_of_yojson (`String uri) in
+  let open Fiber.Result.O in
+  let* doc = Fiber.return (Document_store.get store uri) in
+  let pos = Position.logical position in
+  let open Fiber.O in
+  let+ result =
+    Document.with_pipeline doc (fun pipeline ->
```

Comment:
You can just use `with_pipeline_exn` here. Exceptions are already converted to jsonrpc errors.